node/router/tests/connect.rs

```diff
@@ -150,11 +150,48 @@ async fn test_connect_with_handshake() {
     }
 }
 
+#[tokio::test]
+async fn test_validator_connection() {
+    // Create 2 routers.
+    let node0 = validator(0, 2, false).await;
+    let node1 = validator(0, 2, false).await;
+    assert_eq!(node0.number_of_connected_peers(), 0);
+    assert_eq!(node1.number_of_connected_peers(), 0);
+
+    // Enable handshake protocol.
+    node0.enable_handshake().await;
+    node1.enable_handshake().await;
+
+    // Start listening.
+    node0.tcp().enable_listener().await.unwrap();
+    node1.tcp().enable_listener().await.unwrap();
+
+    {
+        // Connect node0 to node1.
+        node0.connect(node1.local_ip());
+        // Sleep briefly.
+        tokio::time::sleep(Duration::from_millis(200)).await;
```

Comment:
`deadline` could be used here to avoid the `sleep`.

Reply:
Will keep it in mind for future tests, for now I'll keep the logic similar to the rest of the tests in the file.

Reply:
Just tried it. Just awaiting node0's connect is not enough, because node1 might not be connected yet. I suggest we leave refactoring this file for later.